### PR TITLE
Remove duplicate code between Json and Xml read only transaction

### PR DESCRIPTION
**Pull request by JPDSousa**
Summary:
This commit pushes all the encoding agnonstic logic of read-only transactions to
AbstractNodeReadOnlyTrx.

### CODE REVIEW

File: bundles/sirix-core/src/main/java/org/sirix/access/trx/node/xml/XmlNodeReadOnlyTrxImpl.java [hunk 1]
```diff
@@ -95,56 +84,16 @@
...
-    this.trxId = trxId;
-    isClosed = false;
-    itemList = new ItemListImpl();
+    super(trxId, pageReadTransaction, documentNode, resourceManager, new ItemListImpl());
+
   }
 
   @Override
   public boolean storeDeweyIDs() {
+    // TODO should this be resourceManager.getResourceConfig().areDeweyIDsStored?
```
Comment (JPDSousa) [inline bundles/sirix-core/src/main/java/org/sirix/access/trx/node/xml/XmlNodeReadOnlyTrxImpl.java:93]: @JohannesLichtenberger I guess this one's for you

**Comment by JohannesLichtenberger** [comment]
Can you fix the type error? Otherwise a long overdue refactoring -- thanks a lot 👍

**Comment by JPDSousa** [comment]
@JohannesLichtenberger done, tests fixed ✅ Please tell me if those "new" issues (i.e., false positives from moving the code around) are blocking for this to be merged.

**Comment by JohannesLichtenberger** [comment]
No, it's perfect :-) do you like to work on the async commit stuff next?

**Pull request merged**